twine/utils.py

```diff
@@ -211,36 +239,49 @@ def get_username_from_keyring(system):
     except Exception as exc:
         warnings.warn(str(exc))
 
+    return None
 
-def password_prompt(prompt_text):  # Always expects unicode for our own sanity
+
+def password_prompt(prompt_text: str) -> str:
     return getpass.getpass(prompt_text)
 
 
-def get_password_from_keyring(system, username):
+# TODO: See TODOs in get_username_from_keyring
+def get_password_from_keyring(system: str, username: str) -> Optional[str]:
     if 'keyring' not in sys.modules:
-        return
+        return None
 
     try:
-        return sys.modules['keyring'].get_password(system, username)
+        return (
+            # Workaround mypy error `module has no attribute "get_password"`
```

Comment:
It's surprising to me that a workaround would be needed. Keyring unconditionally supplies `get_password` if imported.

Reply:
> It's surprising to me that a workaround would be needed. Keyring unconditionally supplies `get_password` iimported.

I realize this is moot as of #525, but I added a clarifying comment; it looks like mypy doesn't know that the type is `keyring`.

Looks like another option would be `cast(keyring, sys.modules['keyring'])`.